src/store/SwapMetaView.cc

```diff
@@ -0,0 +1,101 @@
+        return;
+    }
+
+    if (DeprecatedSwapMetaType(type)) {
+        debugs(20, DBG_CRITICAL, "ERROR: Ignoring swap meta field with a deprecated type: " << int(type));
+        return;
+    }
+
+    // TODO: Instead of assuming that all future swap meta types can be ignored
+    // (and some should be reported at level-0/1), future-proof this code by
+    // using a type bit to define whether to silently ignore a swap meta field
+    // with that type (or even the whole Store entry with that field).
+
+    const auto typeValuesWeMightAdd = 10;
+    // compute "type > RawSwapMetaTypeTop() + typeValuesWeMightAdd" w/o overflow
+    if (type >= typeValuesWeMightAdd && type - typeValuesWeMightAdd > RawSwapMetaTypeTop()) {
+        debugs(20, DBG_CRITICAL, "ERROR: Malformed cache storage; ignoring swap meta field with an unexpected type: " << int(type));
+        return;
+    }
+
+    if (type > RawSwapMetaTypeTop()) {
+        debugs(20, 3, "ignoring swap meta field with a presumed future type: " << int(type));
+        return;
+    }
+
+    Assure(type <= RawSwapMetaTypeBottom);
+    debugs(20, DBG_CRITICAL, "ERROR: Malformed cache storage; ignoring swap meta field with an invalid type: " << int(type));
+}
+
+} // namespace Store
+
+Store::SwapMetaView::SwapMetaView(const void * const begin, const void * const end)
+{
+    auto input = static_cast<const char *>(begin);
+
+    SwapMetaExtract(rawType, input, end);
+    if (HonoredSwapMetaType(rawType))
+        type = static_cast<SwapMetaType>(rawType);
+    else
+        HandleBadRawType(rawType); // and leave type as STORE_META_VOID
+
+    RawSwapMetaLength lengthOrGarbage = 0;
+    SwapMetaExtract(lengthOrGarbage, input, end);
+    if (lengthOrGarbage < 0)
+        throw TextException("negative swap meta field length value", Here());
+    if (uint64_t(lengthOrGarbage) > SwapMetaFieldValueLengthMax)
+        throw TextException("huge swap meta field length value", Here());
+    if (input + lengthOrGarbage > end)
+        throw TextException("truncated swap meta field value", Here());
+    rawLength = static_cast<size_t>(lengthOrGarbage);
+
+    Assure(input > begin);
+    Assure(input + rawLength <= end);
+    rawValue = input;
+}
+
+void
+Store::SwapMetaView::checkExpectedLength(const size_t expectedLength) const
+{
+    if (rawLength != expectedLength)
+        throw TextException(ToSBuf("Bad value length in a Store entry meta field expecting a ",
+                                   expectedLength, "-byte value: ", *this), Here());
+}
+
+std::ostream &
+operator <<(std::ostream &os, const Store::SwapMetaView &meta)
+{
+    os << "type=" << int(meta.rawType);
+    // XXX: Change Raw constructor to take void* data instead of casting here.
+    const auto rawValue = reinterpret_cast<const char*>(meta.rawValue);
+    // TODO: Add/use something like Raw::bestPresentationEncoding() to report
+    // binary data as hex, URLs as plain text, and Vary with \r\n escapes?
```

Comment:
> IMO this comment should be removed

Removed in b9bbf81 in hope to make progress.

> Any such feature should have a design discussion elsewhere before adding

A TODO does not prohibit any discussions anywhere. The question mark in the TODO already points to a low level of confidence that it should be done. If the developer working on that feature decides there is something that should be discussed before posting a PR, the removed TODO would not stop them from discussing anything they want, anywhere they find appropriate.

> and seems not to be relevant for this code.

It is as relevant as it gets -- this code is printing plain text URLs as hex (for example) because this code does not want to deal with the complexities the removed TODO would address. This is a perfect use case for the feature described in the removed TODO. That does not mean that the feature itself should be added, of course, but it supplies the right context.